client/signup/steps/design-type-with-store/pressable-store/index.jsx

```diff
@@ -20,32 +21,29 @@ import FormLabel from 'components/forms/form-label';
 import FormSectionHeading from 'components/forms/form-section-heading';
 import StepHeader from 'signup/step-header';
 import Button from 'components/button';
-import analytics from 'lib/analytics';
+import { localize } from 'i18n-calypso';
+import { recordTracksEvent } from 'state/analytics/actions';
 
 import HeroImage from './hero-image';
 
-export default React.createClass( {
-	displayName: 'PressableStoreStep',
+class PressableStoreStep extends Component {
+	constructor(props) {
+		super(props);
 
-	propTypes: {
-		onBackClick: PropTypes.func.isRequired,
-	},
-
-	getInitialState() {
-		return {
+		this.state = {
 			email: '',
 			isValid: false,
 			error: null,
 		};
-	},
+	}
 
-	focus() {
+	focus = () => {
 		if ( this._input ) {
 			this._input.focus();
 		}
-	},
+	};
 
-	onEmailChange( event ) {
+	onEmailChange = ( event ) => {
```

Comment:
when `event` isn't otherwise used, I don't see much need to pollute the namespace with a variable for it; you might find this attractive…

```js
onEmailChange = ( { target: { value: email } } ) => {
	const isValid = EmailValidator.validate( email );
	…
}
```